core/src/dht/dht_reducers.rs

```diff
@@ -52,9 +53,68 @@ where
     }
 }
 
+//
+pub(crate) fn commit_sys_entry<CAS, EAVS>(
+    old_store: &DhtStore<CAS, EAVS>,
+    entry: &Entry,
+) -> Option<DhtStore<CAS, EAVS>>
+where
+    CAS: ContentAddressableStorage + Sized + Clone + PartialEq,
+    EAVS: EntityAttributeValueStorage + Sized + Clone + PartialEq,
+{
+    // Add it local storage
+    let mut new_store = (*old_store).clone();
+    let res = new_store.content_storage_mut().add(entry);
+    if res.is_err() {
+        // TODO #439 - Log the error. Once we have better logging.
+        return None;
+    }
+    // Note: System entry types are not published to the network
+    Some(new_store)
+}
+
+//
+pub(crate) fn commit_app_entry<CAS, EAVS>(
+    context: Arc<Context>,
+    old_store: &DhtStore<CAS, EAVS>,
+    entry_type: &EntryType,
+    entry: &Entry,
+) -> Option<DhtStore<CAS, EAVS>>
+where
+    CAS: ContentAddressableStorage + Sized + Clone + PartialEq,
+    EAVS: EntityAttributeValueStorage + Sized + Clone + PartialEq,
+{
+    // pre-condition: if app entry_type must be valid
+    // get entry_type definition
+    let dna = context
+        .state()
+        .unwrap()
+        .nucleus()
+        .dna()
+        .expect("Must have DNA to commit an app entry.");
+    let maybe_def = dna.get_entry_type_def(&entry_type.to_string());
+    if maybe_def.is_none() {
+        return None;
+    }
+    let entry_type_def = maybe_def.unwrap();
+
+    // Add it to local storage...
+    let mut new_store = (*old_store).clone();
+    let res = new_store.content_storage_mut().add(entry);
```

Comment:
Hm, I'm not sure if we should the entry to the local DHT shard at all if it is not public..
@zippy, what to do you say?

Reply:
That's what I implemented in the go version, i.e. that if it's private that the app developer knows they need to get from the chain, and there's a flag in the `get_entry` call for that case, that looks for the hash in the chain only.   I think this probably matters because if it gets into the DHT shard in the Holo case it may mean that the data actually travels outside of a security boundary that it shouldn't.

Reply:
I'm a bit lost here. What needs changing for this to get approved and merged in?

Reply:
If the entry is not public it should not even be stored in the DHT shard. So these two lines should go down into the block guarded by the `if entry_type_def.sharing.clone().can_publish()`.

Reply:
Okay. And can system entries be published ?

Reply:
Yes. All system entries get published except for the Dna, @ddd-mtl.

Reply:
Thanks! This is the sort of info that I think belongs to a spec.

Reply:
#spec